### PR TITLE
Docs: updates for local development

### PR DESCRIPTION
**Pull request by robbymilo**
**What this PR does / why we need it**:

Updates on how the docs build locally for contributors. @oddlittlebird I've updated the docs website image, can you see if this PR gets the website running locally?

### CODE REVIEW

**Comment by oddlittlebird** [comment]
@daniellee did some testing and found there is a dependency not listed in the documentation: you need to have the `grafana/website` repo in order to make the build work.

For some reason, I cannot get that repo to clone onto my Linux VM. I ran the command regularly, got the same Makefile errors. Daniel and I have a meeting scheduled next week to try and sort that out.

I did not test it on Windows, because the command still depends on having `make` installed, and Make does not do well on Windows.

**Comment by aknuds1** [comment]
Why are the changes to Makefile targets `docs` and `docs-no-pull` necessary?

File: docs/Makefile [hunk 1]
```diff
@@ -2,11 +2,14 @@
...
 docs-no-pull:
-	docker run -v $(PWD)/sources:/hugo/content/docs/grafana/latest -p 3002:3002 --rm -it grafana/docs-base:latest /bin/bash -c 'make docs-menu && hugo server -p 3002 -D --ignoreCache --baseUrl http://localhost:3002 --bind 0.0.0.0'
+	docker run -v $(PWD)/sources:/hugo/content/docs/grafana/latest -p 3002:3002 --rm -it grafana/docs-base:latest /bin/bash -c 'mkdir -p content/docs/grafana/latest/ && touch content/docs/grafana/latest/menu.yaml && make docs-menu && hugo server -p 3002 -D --ignoreCache --baseUrl http://localhost:3002 --bind 0.0.0.0'
 
 docs-test:
 	docker pull grafana/docs-base:latest
 	docker run -v $(PWD)/sources:/hugo/content/docs/grafana/latest --rm -it grafana/docs-base:latest /bin/bash -c 'npm i && make prod'
+
+docs-windows:
+	docker run -v c:\GitProjects\grafana\docs\sources:/hugo/content/docs/grafana/latest -p 3002:3002 --rm -it grafana/docs-base:latest /bin/bash -c "mkdir -p content/docs/grafana/latest/ && touch content/docs/grafana/latest/menu.yaml && make docs-menu && hugo server -p 3002 -D --ignoreCache --baseUrl http://localhost:3002 --bind 0.0.0.0"
```
Comment (aknuds1) [inline docs/Makefile]: Won't this depend on the user having Grafana checked out in c:\GitProjects\grafana?

Comment (robbymilo) [reply docs/Makefile]: Nope, that is what the docker image is for.

**Comment by robbymilo** [comment]
> Why are the changes to Makefile targets `docs` and `docs-no-pull` necessary?

They are necessary as some of the commands were giving errors in certain situations.

@oddlittlebird Can you confirm this is now working?

File: docs/Makefile (hunk 1 above)
Comment (oddlittlebird) [reply docs/Makefile]: For Windows, yes. Remember, we got that path by copying it from my personal C drive. That path has nothing to do with Docker.

In order for a Windows user to make this command work, they need to replace that path with the path for their own personal Grafana repo on their machine.

Comment (oddlittlebird) [reply docs/Makefile]: We had to make two major changes to get this to work on my system:
* Change the single quotes to double quotes.
* Replace PWD with my path to the Grafana docs > sources folder.

**Comment by oddlittlebird** [comment]
> > Why are the changes to Makefile targets `docs` and `docs-no-pull` necessary?
> 
> They are necessary as some of the commands were giving errors in certain situations.
> 
> @oddlittlebird Can you confirm this is now working?

Yes, can confirm it worked on my master branch. It did not work on another branch, but that might be because I have a broken link in there somewhere. (Yay for being able to locally check broken links!)

However, while it sees changes to documents, I still need to manually trigger the documentation to rebuild. It will not do so automatically.

File: docs/Makefile (hunk 1 above)
Comment (aknuds1) [reply docs/Makefile]: @robbymilo as @oddlittlebird says, c:\GitProjects\grafana\docs\sources gets mounted from the host machine into the Docker container. Therefore that directory has to be present when issuing the command. Compare to the Unix version of the command where you use `$(PWD)` in order to get the current directory automatically.

Isn't there a method on Windows for getting the current directory, similar to `$(PWD)` on Unix, so we don't hardcode the user's Grafana checkout path?

**Pull request closed without merging**